### ISSUE TITLE
parallel build warning

### ISSUE THREAD

**Issue by andreacassioli** [mcmtroffaes/sphinxcontrib-bibtex#80]
Hi,
after upgrading to Sphinx 1.3 I got the following

     sphinxcontrib.bibtex extension does not declare if it is safe for parallel reading, assuming it isn't -   please ask the extension author to check and make it explicit

In my understanding now extension's setup function  must return information if they can be run in parallel. Any plan on this?

**Comment by mcmtroffaes** [comment]
Indeed...

http://sphinx-doc.org/extdev/index.html#extension-metadata

... I have not yet caught up with this. The extension *should* be safe for reading multiple documents in parallel although it would be worth to run some tests against it. I'll try to look into this later this week but feel free to post a patch in the mean time (should be trivial, and travis should catch any issues).

**Comment by andreacassioli** [comment]
Hi.
thank you for the quick feed back!

Looking around it seems just a line, but I have no idea how to make tests, so I'd rather prefer to keep updates to people more skilled than me!

**Comment by mcmtroffaes** [comment]
Parallel builds are now enabled in the feature/fix-tests branch. Could you test your build with that branch?

For the record, the work on this can be tracked under pull request #81.

**Comment by mcmtroffaes** [comment]
I noticed some unresolved citation issues with parallel reads, so I'll disable them.

**Comment by mcmtroffaes** [comment]
The fix is now merged into develop. Thanks again for reporting!

**Comment by andreacassioli** [comment]
Hi,
thank you a lot for such a prompt reply!

**Comment by mcmtroffaes** [comment]
You're most welcome. The fix is included the latest release on pypi.

**Comment by andreacassioli** [comment]
Hi,
Is the fix out there? I have just realised I still get

   WARNING: the sphinxcontrib.bibtex extension is not safe for parallel reading, doing serial read

I am running on Sphinx 1.3.1, sphinxcontrib-bibtex 0.3.2 freshly installed from pip, python3 on Linux.

**Comment by mcmtroffaes** [comment]
What you see is expected. In fact this behaviour is even tested against: https://github.com/mcmtroffaes/sphinxcontrib-bibtex/blob/develop/test/test_issue80_parallel.py

The extension reports to Sphinx that it cannot do parallel reads, because the bibliography directive must be processed after citations and there is otherwise no way to control this. This is also related to https://github.com/mcmtroffaes/sphinxcontrib-bibtex/issues/8

**Comment by mcmtroffaes** [comment]
... so in short: don't ask sphinx to do parallel reads if you use the bibtex extension if you don't want to see the warning.

**Comment by mcmtroffaes** [comment]
Note this is now being worked on, see #210.
